ring/ring_test.go

```diff
@@ -1843,11 +1843,6 @@ func TestRing_ShuffleShardWithLookback_CorrectnessWithFuzzy(t *testing.T) {
 				t.Run(testName, func(t *testing.T) {
 					t.Parallel()
 
-					// Randomise the seed but log it in case we need to reproduce the test on failure.
-					seed := time.Now().UnixNano()
-					rand.Seed(seed)
-					t.Log("random generator seed:", seed)
-
```

Comment:
Instead of dropping the seeding, can you create a seeded generator instead (also make sure to update rand.* calls below)?

```suggestion
					// Randomise the seed but log it in case we need to reproduce the test on failure.
					seed := time.Now().UnixNano()
					rnd := rand.New(rand.NewSource(seed))
					t.Log("random generator seed:", seed)
```

Reply:
Good point, done.